app/controllers/users_controller.rb

```diff
@@ -1,3 +1,29 @@
 class UsersController < ApplicationController
-  def new; end
+  def new
+    @user = User.new
+  end
+
+  def show
+    @user = User.find_by id: params[:id]
+    return if @user
+
+    flash[:danger] = t ".fail_message"
+    redirect_to root_path
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      flash[:success] = t ".welcome_message"
+      redirect_to @user
+    else
+      render :new
+    end
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit User::USERS_PARAMS
```

Comment:
require có tác dụng gì em? permit có tác dụng gì?
chặn nó lại in params.require(:user) ra đây cho anh

Reply:
method require sẽ tìm params[: user] và sẽ raise lên lỗi nếu không tìm thấy.
method permit định nghĩa 1 danh sách các thuộc tính cho phép.

Reply:
![Screenshot from 2020-08-03 16-18-32](https://user-images.githubusercontent.com/68638634/89167014-0fd29300-d5a5-11ea-846c-8094db401086.png)